util/http.py

```diff
@@ -325,31 +334,29 @@ def debuggable_post(cls, url, payload, **kwargs):
         return cls.debuggable_request("POST", url, **kwargs)
 
     @classmethod
-    def debuggable_request(cls, http_method, url, **kwargs):
+    def debuggable_request(cls, http_method, url, make_request_with=None,
+                           **kwargs):
         """Make a request that returns a detailed problem detail document on
         error, rather than a generic "an integration error occured"
         message.
+
+        :param http_method: HTTP method to use when making the request.
+        :param url: Make the request to this URL.
+        :param make_request_with: A function that actually makes the
+            HTTP request.
+        :param kwargs: Keyword arguments for the make_request_with
+            function.
         """
-        if 'allowed_response_codes' in kwargs:
-            # The caller wants to treat a specific set of response codes
-            # as successful.
-            allowed_response_codes = kwargs['allowed_response_codes']
-        else:
-            # We want request_with_timeout to allow through all
-            # response codes so that we can handle bad ones in a more
-            # helpful way.
-            kwargs['allowed_response_codes']=["1xx", "2xx", "3xx", "4xx", "5xx"]
-
-            # But we want to apply the normal rules when deciding whether
-            # a given response is 'bad'.
-            allowed_response_codes = None
```

Comment:
We can remove this code because we are using `process_debuggable_response` to handle the response, where previously we were calling `request_with_timeout` (which called `process_response`, enforcing the rules of that method) and _then_ calling `process_debuggable_response`.